code/modules/food_and_drinks/machinery/stove_component.dm

```diff
@@ -97,6 +98,18 @@
 /datum/component/stove/proc/on_attack_hand_secondary(obj/machinery/source)
 	SIGNAL_HANDLER
 
+	toggle_mode()
+
+	return COMPONENT_SECONDARY_CANCEL_ATTACK_CHAIN
+
+/datum/component/stove/proc/on_attack_robot_secondary(obj/machinery/source)
+	SIGNAL_HANDLER
+
+	toggle_mode()
+
+	return COMPONENT_SECONDARY_CANCEL_ATTACK_CHAIN
+
+/datum/component/stove/proc/toggle_mode()
```

Comment:
`toggle_mode()` still has a `return COMPONENT_SECONDARY_CANCEL_ATTACK_CHAIN` that you need to address